app/permissions/concerns/decidim/reporting_proposals/admin/permissions_override.rb

```diff
@@ -7,8 +7,81 @@ module PermissionsOverride
         extend ActiveSupport::Concern
 
         included do
+          # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+          def permissions
+            # The public part needs to be implemented yet
+            return permission_action if permission_action.scope != :admin
+
+            # Valuators can only perform these actions
+            if user_is_valuator?
+              if valuator_assigned_to_proposal?
+                can_create_proposal_note?
+                can_create_proposal_answer?
+                allow! if action_is_show_on_proposal?
+              elsif action_is_show_on_proposal?
+                disallow!
+              end
+              can_export_proposals?
+              valuator_can_unassign_valuator_from_proposals?
```

Comment:
I think  we shouldn't override the whole `permissions` method, is too cumbersome.
I we can just work on the override of `valuator_can_unassign_valuator_from_proposals?` and put all the cases there it would be much cleaner.